src/libstd/collections/hash/set.rs

```diff
@@ -764,18 +764,27 @@ pub struct Iter<'a, K: 'a> {
     iter: Keys<'a, K, ()>
 }
 
+unsafe impl<'a, K: Sync> Sync for Iter<'a, K> {}
+unsafe impl<'a, K: Sync> Send for Iter<'a, K> {}
```

Comment:
Can these be automatically inferred by applying the traits to the underlying `Keys` iterator?